src/Util/TestListenerTrait.php

```diff
@@ -126,4 +133,55 @@ public function addError($test, \Exception $e, $time)
             $r->setValue($e, 'Polyfills enabled, '.$r->getValue($e));
         }
     }
+
+    /**
+     * @param string|string[] $files
+     */
+    private static function scanFileForInvalidSignatures($file)
+    {
+        $betterReflection = new BetterReflection();
+        $locator = new SingleFileSourceLocator($file, $betterReflection->astLocator());
+        $reflector = new FunctionReflector($locator, $betterReflection->classReflector());
+        $polyfills = $reflector->getAllFunctions();
+        $warnings = array();
+        foreach ($polyfills as $polyfill) {
+            $function = $polyfill->getName();
+            if (!\function_exists($function)) {
+                continue;
+            }
+
+            $originalFunction = new \ReflectionFunction($function);
+
+            $warnings[] = static::compareParameters($function, $originalFunction->getParameters(), $polyfill->getParameters());
+        }
+
+        return array_filter($warnings);
+    }
+
+    private static function compareParameters($methodOrFunction, $original, $polyfill)
+    {
+        $callable = function($parameter) { return $parameter->getName(); };
+        $polyfillParameters = array_map($callable, $polyfill);
+        $originalParameters = array_map($callable, $original);
+        if ($polyfillParameters !== $originalParameters) {
+            $toString = function($parameters) {
+                return implode(', ', array_map(function($parameter) {
+                    return '$'.$parameter;
+                }, $parameters));
+            };
+
+            $polyfillParametersString = $toString($polyfillParameters);
+            $originalParametersString = $toString($originalParameters);
+
+            return TestListener::warning(<<<FMT
```

Comment:
is a warning the right thing here? or should it be an error/failure?

Reply:
a failure would be the best